tensorboard/backend/event_processing/event_file_loader.py

```diff
@@ -27,6 +29,26 @@
 logger = tb_logging.get_logger()
 
 
+@contextlib.contextmanager
+def _null_context():
+  """Pre-Python-3.7-compatible standin for contextlib.null_context."""
+  yield
+
+
+# Might as well make this a singleton.
+_NULL_CONTEXT = _null_context()
+
+
+def _silence_deprecation_warnings():
+  """Context manager that best-effort silences TF deprecation warnings."""
+  try:
+    # Learn this one weird trick to make TF deprecation warnings go away.
+    from tensorflow.python.util import deprecation
+    return deprecation.silence()
+  except (ImportError, AttributeError):
+    return _NULL_CONTEXT
```

Comment:
You could avoid the `_NULL_CONTEXT` backport with something like this:

```python
@contextlib.contextmanager
def _silence_deprecation_warnings():
  try:
    # Learn this one weird trick to make TF deprecation warnings go away.
    from tensorflow.python.util import deprecation
    cm = deprecation.silence()
  except (ImportError, AttributeError):
    yield
  else:
    with cm:
      yield
```

(Or `cm = None; ...; if cm is not None: ...` if you dislike the `else`.)

If you prefer it as is, then nit: [`nullcontext`], not `null_context`.

[`nullcontext`]: https://docs.python.org/3/library/contextlib.html#contextlib.nullcontext

Reply:
Fixed spelling.  IMO it's actually simpler and more readable just to use the null context manager and have an either-or rather than nested context managers?

Especially given that it's essentially no code to backport (I didn't even realize there was a stdlib one in any version until I checked on a whim).

Reply:
Okay; fine with me if you prefer this.